app/init/draft_upload_manager/index.test.ts

```diff
@@ -0,0 +1,439 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {ClientResponse, ProgressPromise} from '@mattermost/react-native-network-client';
+import {AppState, AppStateStatus} from 'react-native';
+
+import {addFilesToDraft} from '@actions/local/draft';
+import {PROGRESS_TIME_TO_STORE} from '@constants/files';
+import DatabaseManager from '@database/manager';
+import ServerDataOperator from '@database/operator/server_data_operator';
+import {queryDraft} from '@queries/servers/drafts';
+import TestHelper from '@test/test_helper';
+
+import {exportedForTesting} from '.';
+
+const {DraftUploadManager} = exportedForTesting;
+
+const url = 'baseHandler.test.com';
+const mockClient = TestHelper.createClient();
+
+jest.mock('@init/network_manager', () => {
+    const original = jest.requireActual('@init/network_manager');
+    return {
+        ...original,
+        getClient: (serverUrl: string) => {
+            if (serverUrl === url) {
+                return mockClient;
+            }
+
+            throw new Error('client not found');
+        },
+    };
+});
+
+const now = new Date('2020-01-01').getTime();
+const timeNotStore = now + (PROGRESS_TIME_TO_STORE - 1);
+const timeStore = now + PROGRESS_TIME_TO_STORE + 1;
+
+const mockUpload = () => {
+    const returnValue: {
+        resolvePromise: ((value: ClientResponse | PromiseLike<ClientResponse>) => void) | null;
+        rejectPromise: ((reason?: any) => void) | null;
+        progressFunc: ((fractionCompleted: number, bytesRead?: number | null | undefined) => void) | null;
+    } = {
```

Comment:
A preference here and completely optional - you can further extract those lines into its own type.

Reply:
Since (for now) we are not using that type anywhere else, I would prefer to keep it here.